ddsrouter_core/src/cpp/types/participant/ParticipantKind.cpp

```diff
@@ -30,114 +30,43 @@ namespace ddsrouter {
 namespace core {
 namespace types {
 
-const std::map<ParticipantKindType, std::vector<std::string>> ParticipantKind::participant_kind_with_aliases_ =
-{
-    {PARTICIPANT_KIND_INVALID, {"__invalid_participant_kind__"}},
-    {VOID, {"void"}},
-    {ECHO, {"echo"}},
-    {DUMMY, {"dummy"}},
-    {SIMPLE_RTPS, {"local", "simple"}},
-    {LOCAL_DISCOVERY_SERVER, {"discovery-server", "ds", "local-ds", "local-discovery-server"}},
-    {WAN, {"wan", "router"}},
-};
-
-ParticipantKind::ParticipantKind(
-        ParticipantKindType value) noexcept
-    : value_(value)
-{
-}
-
-ParticipantKind::ParticipantKind() noexcept
-    : ParticipantKind(PARTICIPANT_KIND_INVALID)
-{
-}
-
-bool ParticipantKind::is_valid() const noexcept
-{
-    return value_ != PARTICIPANT_KIND_INVALID;
-}
-
-std::string ParticipantKind::to_string() const noexcept
-{
-    auto it = ParticipantKind::participant_kind_with_aliases_.find(value_);
-
-    // Value must be in map
-    assert(it != ParticipantKind::participant_kind_with_aliases_.end());
-
-    return it->second[0];
-}
-
-ParticipantKindType ParticipantKind::operator ()() const noexcept
-{
-    return value_;
-}
-
-bool ParticipantKind::operator <(
-        const ParticipantKind& other) const noexcept
-{
-    return this->value_ < other.value_;
-}
-
-bool ParticipantKind::operator ==(
-        const ParticipantKind& other) const noexcept
+std::ostream& operator <<(
+        std::ostream& os,
+        ParticipantKind kind)
 {
-    return (*this)() == other();
-}
+    try
+    {
 
-ParticipantKind ParticipantKind::participant_kind_from_name(
-        std::string kind) noexcept
-{
-    // Pass kind name to lowercase
-    utils::to_lowercase(kind);
+        os << ParticipantKindStrings.at(static_cast<ParticipantKindType>(kind));
```

Comment:
NIT: I suppose these blank lines would be removed with uncrustify. If not, please remove them.